File: swift2.3学习/代码/day-2/6-block-oc/6-block-oc/wjHttpTool.m
//
//  wjHttpTool.m
//  6-block-oc
//
//  Created by gouzi on 2017/6/2.
//  Copyright © 2017年 wj. All rights reserved.
//

#import "wjHttpTool.h"

@implementation wjHttpTool

- (void)wjLoadDataWithBlock:(wjLoadDataBlock)block {
    
    dispatch_async(dispatch_get_global_queue(0, 0), ^{
        // 异步全局队列
        NSLog(@"发送网络请求 : %@", [NSThread currentThread]);
        
        dispatch_sync(dispatch_get_main_queue(), ^{
            NSLog(@"拿到数据 ： %@", [NSThread currentThread]);
            block(@"json data is getting");
        });
        
    });
}




@end

File: swift2.3学习/代码/day-2/7-闭包/7-闭包/wjHttpTool.swift
//
//  wjHttpTool.swift
//  7-闭包
//
//  Created by gouzi on 2017/6/2.
//  Copyright © 2017年 wj. All rights reserved.
//

import UIKit

class wjHttpTool: NSObject {
    
    // 定义闭包属性
    var callBack : ((_ jsonData : String)->())?
    
    
    // 闭包类型: (参数列表)->(返回值类型)
    func wjLoadData(callBack : @escaping (_ jsonData : String)->()) {
        self.callBack = callBack // 引起强引用
        DispatchQueue.global().async {
            print("当前的线程是 : \(Thread.current)")
            DispatchQueue.main.sync {
                print("获取到数据,进行回调 : \(Thread.current)")
                callBack("json data")
            }
        }
    }
    
}
